dev-cpp/tbb/patches/tbb-2022.0.0.patchset

```diff
@@ -0,0 +1,230 @@
+index cf6d8bd..12282cb 100644
+--- a/cmake/compilers/GNU.cmake
++++ b/cmake/compilers/GNU.cmake
+@@ -71,7 +71,9 @@ if (CMAKE_SYSTEM_PROCESSOR MATCHES "(AMD64|amd64|i.86|x86)" AND NOT EMSCRIPTEN)
+     set(TBB_COMMON_COMPILE_FLAGS ${TBB_COMMON_COMPILE_FLAGS} -mrtm $<$<AND:$<NOT:$<CXX_COMPILER_ID:Intel>>,$<NOT:$<VERSION_LESS:${CMAKE_CXX_COMPILER_VERSION},11.0>>>:-mwaitpkg>)
+ endif()
+ 
+-set(TBB_COMMON_LINK_LIBS ${CMAKE_DL_LIBS})
++if (NOT HAIKU)
++    set(TBB_COMMON_LINK_LIBS ${CMAKE_DL_LIBS})
++endif ()
+ 
+ # Ignore -Werror set through add_compile_options() or added to CMAKE_CXX_FLAGS if TBB_STRICT is disabled.
+ if (NOT TBB_STRICT AND COMMAND tbb_remove_compile_flag)
+diff --git a/include/oneapi/tbb/detail/_config.h b/include/oneapi/tbb/detail/_config.h
+index e676b15..f1f34ab 100644
+--- a/include/oneapi/tbb/detail/_config.h
++++ b/include/oneapi/tbb/detail/_config.h
+@@ -274,7 +274,7 @@
+     #define __TBB_CPP20_COMPARISONS_PRESENT 0
+ #endif
+ 
+-#define __TBB_RESUMABLE_TASKS                           (!__TBB_WIN8UI_SUPPORT && !__ANDROID__ && !__QNXNTO__ && (!__linux__ || __GLIBC__))
++#define __TBB_RESUMABLE_TASKS                           (!__TBB_WIN8UI_SUPPORT && !__ANDROID__ && !__HAIKU__ && !__QNXNTO__ && (!__linux__ || __GLIBC__))
+ 
+ /* This macro marks incomplete code or comments describing ideas which are considered for the future.
+  * See also for plain comment with TODO and FIXME marks for small improvement opportunities.
+@@ -493,6 +493,11 @@
+     #define TBB_ALLOCATOR_TRAITS_BROKEN 1
+ #endif
+ 
++// HAIKU build fix
++#if defined(__HAIKU__)
++    #define TBB_ALLOCATOR_TRAITS_BROKEN 1
++#endif
++
+ // GCC 4.8 C++ standard library implements std::this_thread::yield as no-op.
+ #if __TBB_GLIBCXX_VERSION >= 40800 && __TBB_GLIBCXX_VERSION < 40900
+     #define __TBB_GLIBCXX_THIS_THREAD_YIELD_BROKEN 1
+diff --git a/src/tbb/allocator.cpp b/src/tbb/allocator.cpp
+index 689c512..559ddf7 100644
+--- a/src/tbb/allocator.cpp
++++ b/src/tbb/allocator.cpp
+@@ -40,7 +40,7 @@
+ // memalign() and it offers nothing but overhead due to inconvenient interface. This is likely the case with other
+ // standard libraries as well, and more libraries can be added to the preprocessor check below. Unfortunately, we
+ // can't detect musl, so we simply enable memalign() on Linux and Android in general.
+-#if defined(linux) || defined(__linux) || defined(__linux__) || defined(__ANDROID__)
++#if defined(linux) || defined(__linux) || defined(__linux__) || defined(__ANDROID__) || defined(__HAIKU__)
+ #include <malloc.h> // memalign
+ #define __TBB_USE_MEMALIGN
+ #else
+@@ -119,7 +119,7 @@ static const dynamic_link_descriptor MallocLinkTable[] = {
+ #define MALLOCLIB_NAME "tbbmalloc" DEBUG_SUFFIX ".dll"
+ #elif __APPLE__
+ #define MALLOCLIB_NAME "libtbbmalloc" DEBUG_SUFFIX ".2.dylib"
+-#elif __FreeBSD__ || __NetBSD__ || __OpenBSD__ || __sun || _AIX || __ANDROID__
++#elif __FreeBSD__ || __NetBSD__ || __OpenBSD__ || __sun || _AIX || __ANDROID__ || __HAIKU__
```

Comment:
Shouldn't Haiku use the `__unix__` case below?

Reply:
Copy/Paste from previous version. must have been a reason back then? maybe fixed by now?

Reply:
It could be a problem with `develop/lib` vs. `lib`, which is unusual.

I don't know what this is actually used for, but we do have versioned .so files, so I thought that it ought to work with the version numbers.

Reply:
Can't remember why it was added, also didn't check if it failed without it. Good thing it's still disabled. :)